# Download de dataset

In [81]:
import sqlite3
import contextlib
import pandas as pd

In [4]:
sql_connect = sqlite3.connect('database.sqlite')
cursor = sql_connect.cursor()

In [39]:
query_r = "SELECT * FROM sqlite_sequence" # read query
query_i = "INSERT INTO country VALUES ('7819', 'mooi land')" # insert query
query_d = "DELETE FROM country WHERE name='mooi land'" # delete query

In [29]:
sql_connect = sqlite3.connect('database.sqlite')
cursor = sql_connect.cursor()
df = pd.read_sql_query(query_r,sql_connect)
df

id         name
0       1      Belgium
1    1729      England
2    4769       France
3    7809      Germany
4   10257        Italy
5   13274  Netherlands
6   15722       Poland
7   17642     Portugal
8   19694     Scotland
9   21518        Spain
10  24558  Switzerland

In [40]:
import contextlib
@contextlib.contextmanager
def db_read_only(file):
    sql_connect = sqlite3.connect('file:'+file+'?mode=ro',uri=True)
    cursor = sql_connect.cursor()
    yield sql_connect
    sql_connect.commit()
    sql_connect.close() 

In [41]:
file = 'database.sqlite'
with db_read_only(file) as ro:
    data = pd.read_sql_query(query_r, ro)

In [42]:
data['name']

0                 Team
1              Country
2               League
3                Match
4               Player
5    Player_Attributes
6      Team_Attributes
Name: name, dtype: object

In [43]:
df_list = []
with db_read_only(file) as ro:
    for table in data['name']:
        query_r = 'SELECT * FROM ' + str(table)
        data = pd.read_sql_query(query_r, ro)
        df_list.append(data)
df_list[0]

id  team_api_id  team_fifa_api_id      team_long_name team_short_name
0        1         9987             673.0            KRC Genk             GEN
1        2         9993             675.0        Beerschot AC             BAC
2        3        10000           15005.0    SV Zulte-Waregem             ZUL
3        4         9994            2007.0    Sporting Lokeren             LOK
4        5         9984            1750.0   KSV Cercle Brugge             CEB
..     ...          ...               ...                 ...             ...
294  49479        10190             898.0       FC St. Gallen             GAL
295  49837        10191            1715.0             FC Thun             THU
296  50201         9777             324.0         Servette FC             SER
297  50204         7730            1862.0  FC Lausanne-Sports             LAU
298  51606         7896               NaN              Lugano             LUG

[299 rows x 5 columns]

In [80]:
team = df_list[0]
country = df_list[1]
league = df_list[2]
match = df_list[3]
player = df_list[4]
player_attributes = df_list[5]
team_attributes = df_list[6]

# Kies een club en bijbehorende competitie

- Club: Atlético Madrid
- Competitie: Spain LIGA BBVA

# Beschrijf alle tabellen en alle data uit de tabellen

Team: In deze tabel is informatie te vinden over alle Europese teams waar data van is verzameld.

Country: In deze tabel staan alle landen waar data van is verzameld.

League: In deze tabel zijn alle verschillende competities te vinden, per land.

Match: In deze tabel zijn de gegevens van alle gespeelde wedstrijden.

Player: In deze tabel zijn alle persoonlijke gegevens te vinden van de spelers van de clubs.

Player Attributes: In deze tabel zijn gegevens te vinden van de spelers die te maken hebben met de manier waarop ze voetballen. 

Team Attributes: In deze tabel staan statestieken over de voetbalclubs.

# Bepaal de gegevens van jouw club

In [68]:
team_info = team[team['team_long_name'] == "Atlético Madrid"]
team_info

id  team_api_id  team_fifa_api_id   team_long_name team_short_name
269  43053         9906             240.0  Atlético Madrid             AMA

- Team name = Atlético Madrid
- Team id = 9906

In [79]:
league_info = league[league['name'] == 'Spain LIGA BBVA']
league_info

id  country_id             name
9  21518       21518  Spain LIGA BBVA

- Competitie naam = Spain LIGA BBVA
- Competitie id = 21518

Gegevens van alle gespeelde wedstrijden door onze club.

In [77]:
match[(match['home_team_api_id'] == 9906) | (match['away_team_api_id'] == 9906)]

id  country_id  league_id     season  stage                 date  \
21526  21527       21518      21518  2008/2009      1  2008-08-31 00:00:00   
21531  21532       21518      21518  2008/2009     10  2008-11-09 00:00:00   
21540  21541       21518      21518  2008/2009     11  2008-11-16 00:00:00   
21555  21556       21518      21518  2008/2009     12  2008-11-23 00:00:00   
21560  21561       21518      21518  2008/2009     13  2008-11-30 00:00:00   
...      ...         ...        ...        ...    ...                  ...   
24510  24511       21518      21518  2015/2016      5  2015-09-22 00:00:00   
24521  24522       21518      21518  2015/2016      6  2015-09-26 00:00:00   
24531  24532       21518      21518  2015/2016      7  2015-10-04 00:00:00   
24542  24543       21518      21518  2015/2016      8  2015-10-18 00:00:00   
24552  24553       21518      21518  2015/2016      9  2015-10-25 00:00:00   

       match_api_id  home_team_api_id  away_team_api_id  home_team_goal  ...  \
21526        530092              9906              9864               4  ...   
21531        530237              8371              9906               0  ...   
21540        530246              9906              9783               4  ...   
21555        530261              8388              9906               1  ...   
21560        530326              9906              8696               4  ...   
...             ...               ...               ...             ...  ...   
24510       2030125              9906              8305               2  ...   
24521       2030136             10205              9906               1  ...   
24531       2030146              9906              8633               1  ...   
24542       2030157              8560              9906               0  ...   
24552       2030167              9906             10267               2  ...   

       SJA   VCH   VCD    VCA   GBH   GBD   GBA   BSH   BSD  BSA  
21526  9.0  1.45  3.75   7.00  1.50  3.60  7.50  1.50  3.75  6.0  
21531  2.1  2.90  3.20   2.25  3.10  3.25  2.25  3.00  3.25  2.2  
21540  6.0  1.60  3.60   5.50  1.67  3.50  5.20  1.67  3.40  5.0  
21555  2.0  3.60  3.30   2.00  3.60  3.20  2.05  3.40  3.25  2.1  
21560  6.0  1.55  3.75   5.50  1.60  3.60  5.75  1.62  3.40  5.5  
...    ...   ...   ...    ...   ...   ...   ...   ...   ...  ...  
24510  NaN  1.29  5.75  15.00   NaN   NaN   NaN   NaN   NaN  NaN  
24521  NaN  2.88  3.20   2.75   NaN   NaN   NaN   NaN   NaN  NaN  
24531  NaN  3.13  3.40   2.45   NaN   NaN   NaN   NaN   NaN  NaN  
24542  NaN  4.30  3.30   2.05   NaN   NaN   NaN   NaN   NaN  NaN  
24552  NaN  1.57  4.00   7.00   NaN   NaN   NaN   NaN   NaN  NaN  

[304 rows x 115 columns]